### ISSUE TITLE
Loadouts Apply Wrong Items/Shuffle Inventory

### ISSUE THREAD

**Issue by 4xle** [DestinyItemManager/DIM#5815]
DIM Version: 6.30.0(release)
Browsers: Firefox/Chrome
Beta Version: No

Expected Behavior: Saved Loadouts apply as listed when clicked.
Encountered Behavior: Either nothing happens, though DIM shows a loadout as successfully applied 
OR
Gear is changed, but to seemingly random gear which is not the gear listed in the loadout.

It doesn't seem to matter if "Move Items Away" is checked or not, as the inventory items are changed to some degree as well, though it is difficult to tell if changes are happening in bulk or with singular items(currently not enough space to vault everything and test more thoroughly).

These loadouts were custom loadouts made before the Loadout Optimizer feature was implemented if that makes any difference.

**Comment by bhollis** [comment]
Hmm. Loadout Optimizer was built several years ago - maybe you should try recreating the loadouts and see if that helps? Though I don't know why that'd cause that behavior.

**Comment by bhollis** [comment]
Did this just start happening in the most recent release that went out Sunday?

**Comment by bhollis** [comment]
Can you provide screenshots of your inventory before and after applying the loadout, and of your loadout's contents?

**Comment by 4xle** [comment]
Whoops, typo earlier. I meant that the Loadouts were definitely made before the loadout "Comparison" feature was released. My bad.

Images of operations performed in order attached with the builds between them. Both builds were made brand new for testing purposes.

![StartingPoint](https://user-images.githubusercontent.com/23562196/93284546-8c59b380-f798-11ea-95d5-7bd80ed65a96.png)
![AttemptedApplyBuild](https://user-images.githubusercontent.com/23562196/93284557-9085d100-f798-11ea-9a7a-b414670783ad.png)
![AttemptedLoadout](https://user-images.githubusercontent.com/23562196/93284562-92e82b00-f798-11ea-9ade-605d759aada1.png)
![StartingPointBuild](https://user-images.githubusercontent.com/23562196/93284584-9da2c000-f798-11ea-93cc-8dbf23375800.png)
![AttemptedReturnToStartingPointAndGoesHorriblyWrong](https://user-images.githubusercontent.com/23562196/93284589-9f6c8380-f798-11ea-981b-b17486f954f4.png)

**Comment by bhollis** [comment]
Can you show what is printed in the developer console when you apply one of the loadouts?

**Comment by 4xle** [comment]
The console does not appear to print anything consistently when applying loadouts. Occasionally mentioned toggling locking state, but I've not been able to repro that since I saw it last. I tried applying several loadouts, no logs were written beyond the initial page load, as far as I can see?
```
Sync: Storage may be cleared under storage pressure.
sync.service.ts:55 Sync: DIM is using 114.22 MB total out of 1.09 TB in storage quota (0%).
Index.tsx:48 DIM v6.30.0 (release) - Please report any errors to https://www.github.com/DestinyItemManager/DIM/issues
platforms.ts:19 Load accounts from IDB
actions.ts:116 globalSettings {dimApiEnabled: true, destinyProfileMinimumRefreshInterval: 15, destinyProfileRefreshInterval: 30, autoRefresh: false, refreshProfileOnVisible: true, …}
3authenticated-fetch.ts:121 Successfully updated auth token from refresh token.
manifest-service-json.ts:108 Timer 'Load manifest' already exists
(anonymous) @ manifest-service-json.ts:108
(anonymous) @ index.js:8
dispatch @ redux.js:638
(anonymous) @ manifest-service-json.ts:96
(anonymous) @ util.ts:80
(anonymous) @ index.js:8
dispatch @ redux.js:638
(anonymous) @ d2-definitions.ts:152
(anonymous) @ index.js:8
(anonymous) @ d2-stores.ts:196
(anonymous) @ loading-tracker.ts:27
(anonymous) @ switchMap.ts:122
(anonymous) @ Subscriber.ts:100
(anonymous) @ mergeMap.ts:170
(anonymous) @ innerSubscribe.ts:31
(anonymous) @ Subscriber.ts:100
(anonymous) @ Subject.ts:71
(anonymous) @ BehaviorSubject.ts:44
getStoresStream @ d2-stores.ts:166
(anonymous) @ hooks.ts:20
ou @ react-dom.production.min.js:211
ys @ react-dom.production.min.js:257
(anonymous) @ scheduler.production.min.js:19
Wi @ react-dom.production.min.js:122
gs @ react-dom.production.min.js:257
(anonymous) @ react-dom.production.min.js:256
z @ scheduler.production.min.js:17
(anonymous) @ scheduler.production.min.js:14
manifest-service-json.ts:141 Load manifest: 2039.60595703125 ms
actions.ts:78 Saving profile data to IDB
manifest-service-json.ts:141 Timer 'Load manifest' does not exist
(anonymous) @ manifest-service-json.ts:141
async function (async)
(anonymous) @ manifest-service-json.ts:109
(anonymous) @ index.js:8
dispatch @ redux.js:638
(anonymous) @ manifest-service-json.ts:96
(anonymous) @ util.ts:80
(anonymous) @ index.js:8
dispatch @ redux.js:638
(anonymous) @ d2-definitions.ts:152
(anonymous) @ index.js:8
(anonymous) @ d2-stores.ts:196
(anonymous) @ loading-tracker.ts:27
(anonymous) @ switchMap.ts:122
(anonymous) @ Subscriber.ts:100
(anonymous) @ mergeMap.ts:170
(anonymous) @ innerSubscribe.ts:31
(anonymous) @ Subscriber.ts:100
(anonymous) @ Subject.ts:71
(anonymous) @ BehaviorSubject.ts:44
getStoresStream @ d2-stores.ts:166
(anonymous) @ hooks.ts:20
ou @ react-dom.production.min.js:211
ys @ react-dom.production.min.js:257
(anonymous) @ scheduler.production.min.js:19
Wi @ react-dom.production.min.js:122
gs @ react-dom.production.min.js:257
(anonymous) @ react-dom.production.min.js:256
z @ scheduler.production.min.js:17
(anonymous) @ scheduler.production.min.js:14
d2-stores.ts:256 Process inventory: 135.787109375 ms
d2-stores.ts:260 Inventory state update: 917.80810546875 ms
d2-stores.ts:256 Process inventory: 129.260986328125 ms
d2-stores.ts:260 Inventory state update: 499.051025390625 ms
d2-stores.ts:256 Process inventory: 185.2958984375 ms
d2-stores.ts:260 Inventory state update: 221.24609375 ms
```

**Comment by bhollis** [comment]
Only on Beta

**Comment by 4xle** [comment]
I'll give it a try on beta today and see if anything useful jumps out from the logs.
________________________________

**Comment by 4xle** [comment]
Hahaha okay, you ready for this? I managed to get lucky at one point with the beta app, everything working exactly like I expected, reading the console I see the message for "apply loadout to Vault". Think to myself: that's not what I did. Is it? And it wasn't. But an addon I have for webdev work was active, and it was intercepting and passing clicks to the vault loadout calls due to a collision in class names (short story version). So. This can be closed, because the app is fine and awesome. I'm going to eat my user-error flavored crayons now and remember to ensure my work addons are disabled.
